FogCreek/Glitch-Community
# Ancient salmonberry: Convert Categories to Collections

UPDATE! this pr has changed a bunch since I wrote this description. This should now be a 1:1 replacement of categories to collections, and should hopefully not be noticeable to users

## Links
* Remix link: https://ancient-salmonberry.glitch.me/
* Issue-tracker link:
    - https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/528
    - https://app.zenhub.com/workspaces/community-5d07d1359dee6059a9688176/issues/fogcreek/glitch-community-work/525
* Specs/doc links: https://docs.google.com/document/d/127bu23-WBPQmkFWMuFCYyc43wvtoYh_EokvLtRT3uQY/edit#

## GIF/Screenshots:
![Screen Shot 2019-10-11 at 5 15 01 PM](https://user-images.githubusercontent.com/6620164/66685624-12957d80-ec4b-11e9-91e2-fb123f458540.png)
![Screen Shot 2019-10-11 at 5 15 26 PM](https://user-images.githubusercontent.com/6620164/66685625-12957d80-ec4b-11e9-8bf6-25850ff45e46.png)

## Changes:
* All previous categories pages (ex: glitch.com/music) now use **collection** data instead of category data so that they are easier to manage by dev advocates. This will also fix an existing bug with Categories and My Stuff and help prevent future bugs with the new exploring collections project.
* Combines Art and Music into one category (since there is now going to be one collection for the two of them)
* Removes Learn to Code from the list

## How To Test:
* go to https://ancient-salmonberry.glitch.me/secret and enable the onboarding banners
* go to https://ancient-salmonberry.glitch.me/ and sign in with a new account that does not have any projects
* notice on the homepage that the onboarding banner has the new list of categories, make sure they all go to reasonable locations, try resizing the window to make sure it looks good
* notice on the user page that the onboarding banner has a new list of categories, make sure they also go to reasonable locations, try resizing the window to make sure it looks good
* go to the /create page and scroll to the bottom to the categories list and make sure they all go to reasonable locations, try resizing the window to make sure it looks good
* ensure that all old categories links go to an actively managed collection and it's not a dead link in case we have old tweets or youtubes referring to them:
    - https://ancient-salmonberry.glitch.me/hello-worlds
    - https://ancient-salmonberry.glitch.me/games
    - https://ancient-salmonberry.glitch.me/building-blocks
    - https://ancient-salmonberry.glitch.me/learn-to-code
    - https://ancient-salmonberry.glitch.me/tools-for-work
    - https://ancient-salmonberry.glitch.me/community-picks
    - https://ancient-salmonberry.glitch.me/handy-bots
    - https://ancient-salmonberry.glitch.me/hardware
    - https://ancient-salmonberry.glitch.me/art
    - https://ancient-salmonberry.glitch.me/music

## Feedback I'm looking for:
I'm probably missing something, what am i forgetting about? Can anyone think of something special categories give us that collections doesn't have? Is there any reason not to do this?

Stylistically a lot of these illustrations are not quite matching anymore, but they are not totally off? idk, I'm open to folks saying we need to pause this until we can get illustrations that best match what we're talking about.

## Things left to do before deploying:
- [ ] ideas is currently empty! let's have at least some stuff in there before we deploy this
